### ISSUE TITLE
On Sale Products block: Prevent notice from appearing when clicking on Add to Cart button inside Single Product Template

### ISSUE THREAD

**Issue by thealexandrelara** [woothemes/woocommerce#42156]
## Describe the bug

When the On Sale Products block is inserted to the Single Product template and the shopper clicks on the Add to Cart button, it triggers the notice from the Single Product page.

## To reproduce

1. From your WordPress dashboard, go to Appearance > Themes. Select a blocked theme, such as: "Twenty-twenty four theme"
2. On the left-hand side menu, click on Appearance > Editor > Templates
3. Find and select the 'Single Product' template from the list.
4. When the Classic Product Template renders, click on Transform into Blocks. This will transform the Classic template in a block template if you haven't done it before.
5. On the top-right side, click on the Save button.
6. Add the 'On Sale Products' block.
7. Click on the Save button;
8. Visit a product page and click on Add to Cart button from one of the products inside the On Sale Products block.
9. See that the Notice appears on the page.

## Expected behavior

The notice should not appear when the block is added to the Single Product template and the shopper clicks on Add to Cart button from one of the products inside the On Sale Products block.

## Screenshots

https://github.com/woocommerce/woocommerce-blocks/assets/20469356/f320c0b4-4312-447b-a5c2-f7300ad78834

### WordPress (please complete the following information):

-   WordPress version: 6.4
-   WooCommerce Blocks version: 11.4-dev

**Comment by nerrad** [comment]
Are we _sure_ this is specific to TT4? This doesn't look like something that would be theme-specific (unless the notice appears on other themes too but in a different location)?

**Comment by thealexandrelara** [comment]
> Are we _sure_ this is specific to TT4? This doesn't look like something that would be theme-specific (unless the notice appears on other themes too but in a different location)?

You're right, I just tested with other themes and it's occurring for other block-based themes as well. I changed the title and removed the label, thank you for pointing this out 🙌

**Comment by danieldudzic** [comment]
@thealexandrelara I've tested this and gave it some thought.

This doesn't seem like a bug to me. What should the behavior be? The Store Notice shouldn't be displayed? What other indication should there be that a product was added to the cart?

This behavior seems to be coming from WooCommerce Core and doesn't seem incorrect to me.

The Store Notices just outputs status updates for WooCommerce related things.

A product (no matter what kind) being added to cart seems to warrant a notice.

**Comment by thealexandrelara** [comment]
> @thealexandrelara I've tested this and gave it some thought.
> 
> This doesn't seem like a bug to me. What should the behavior be? The Store Notice shouldn't be displayed? What other indication should there be that a product was added to the cart?
> 
> This behavior seems to be coming from WooCommerce Core and doesn't seem incorrect to me.
> 
> The Store Notices just outputs status updates for WooCommerce related things.
> 
> A product (no matter what kind) being added to cart seems to warrant a notice.

The Store Notices used to be displayed only when shoppers added the visited "single" product to the cart. For example, when visiting a "single" product, consider the Related Products (which is a "default" block for the Single Product Template): when you click on "Add to Cart" for any of the related products, the notice does not appear.

Another example, the Store Notices block also exists within the Product Catalog template, but when you add any products from the Products (Beta) block to the cart, no notice is shown.

If what you mentioned is the expected behavior, it seems that we have multiple "broken" blocks because blocks like Products (Beta), Products Collection, etc., do not trigger the notice to appear on the page when adding products to the cart. It seems that currently, the On Sale Products block is the only one that causes the Store Notices block to be shown (besides the Add to Cart With Options block within the Single Product page)

**Comment by danieldudzic** [comment]
Right, it seems that only blocks not using the Interactivity API Add to Cart buttons (so, `On Sale Products`), which result in a reload, trigger the notice.

So, if we were to address this, migrating the On Sale Products block to use the Interactivity API Add to Cart button would be a solution.

But since this is a legacy block, I don't believe this is high priority enough to warrant such effort.

What do you think, @Aljullu?

**Comment by Aljullu** [comment]
> But since this is a legacy block, I don't believe this is high priority enough to warrant such effort.

:100: from my side. I agree the current behavior is not the best, but as we are focusing our efforts in the Product Collection block and this issue only affects the legacy product grid blocks, I would mark it as low priority.

**Comment by Aljullu** [comment]
I went ahead and added the `priority: low` label. :+1:

**Comment by kkoteen** [comment]
Hi @thealexandrelara - thanks for the suggestion.

I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've reviewed this from a product perspective, and we don't plan to pursue this change in WooCommerce core at this time.

I'm going to close this as not planned. If the context changes or there's a stronger use case we should consider, we can revisit it.
